pallas-txbuilder/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+use pallas_traverse::wellknown::GenesisValues;
+
+mod asset;
+mod builder;
+mod fee;
+mod transaction;
+
+pub mod plutus;
+pub mod prelude;
+
+#[derive(Debug, Clone)]
+pub struct NetworkParams {
+    pub genesis_values: GenesisValues,
+}
+
+impl NetworkParams {
+    pub fn mainnet() -> Self {
+        Self {
+            genesis_values: GenesisValues::mainnet(),
+        }
+    }
+
+    pub fn testnet() -> Self {
+        Self {
+            genesis_values: GenesisValues::testnet(),
+        }
+    }
+
+    pub fn network_id(&self) -> u64 {
+        self.genesis_values.network_id
+    }
+
+    pub fn timestamp_to_slot(&self, timestamp: u64) -> Option<u64> {
+        (timestamp / self.genesis_values.shelley_slot_length as u64)
+            .checked_sub(self.genesis_values.shelley_known_time)
+    }
+}
```

Comment:
Shouldn't this be something more like
```
(timestamp - shelley_time) / slot_length
```
(or the checked version of that)?

This is subtracting a time from a slot I think.

Reply:
True! Fixed.